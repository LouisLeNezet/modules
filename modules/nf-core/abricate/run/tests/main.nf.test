nextflow_process {

    name "Test Process ABRICATE_RUN"
    script "../main.nf"
    process "ABRICATE_RUN"
    tag "modules"
    tag "modules_nfcore"
    tag "abricate"
    tag "abricate/run"

    test("bacteroides_fragilis - genome.fa.gz") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }

    test("bacteroides_fragilis - genome.fa.gz -- stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                ]
                input[1] = []
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }
}
