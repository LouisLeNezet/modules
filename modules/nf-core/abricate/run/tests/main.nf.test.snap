{
    "bacteroides_fragilis - genome.fa.gz -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2204fb00277c287f5f3d82e28964aa03"
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2204fb00277c287f5f3d82e28964aa03"
                ]
            },
            {
                "ABRICATE_RUN": {
                    "abricate": "1.0.1"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-10T22:50:49.287882484"
    },
    "bacteroides_fragilis - genome.fa.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,b48f4f5f78982a221038fd9f0a3aa504"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2204fb00277c287f5f3d82e28964aa03"
                ],
                "report": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,b48f4f5f78982a221038fd9f0a3aa504"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2204fb00277c287f5f3d82e28964aa03"
                ]
            },
            {
                "ABRICATE_RUN": {
                    "abricate": "1.0.1"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-10T22:50:41.585411247"
    }
}
